### ISSUE TITLE
padding between columns

### ISSUE THREAD

**Issue by super-tomcat** [jiro4989/align.nim#2]
It would be useful if we could control the amount of spaces (or padding if you like) between each column, as it is alignment just does its own padding, for example with alignleft we get this, as we can see the first 2 columns do not even end up with any spaces between them and the rest are uneven as well, NOTE, that i have had to write the number of spaces here because the allowed formatting on here was not showing the spaces:

`ニール・ヒギンズ <- 0 SPACES -> 五十六   <- 10 SPACES ->      五     <- 14 SPACES ->        キリスト教徒
`
want i really want to sort this out is supply how many spaces i want between each column for example lets say 8 would give me 8 spaces between each column like this:

`ニール・ヒギンズ   <- 8 SPACES ->    五十六   <- 8 SPACES ->     五    <- 8 SPACES ->    キリスト教徒 <- 8 SPACES ->`

**Comment by jiro4989** [comment]
Like this ?

```nim
# width is new optional parameter
let a = alignLeft(["hello", "he"], width = 10)
let b = alignLeft(["world", "wo"], width = 10)
let c = alignLeft(["foobar", "foo"], width = 10)
for i, v in a:
  echo a[i] & b[i] & c[i]
```

result

```
# 10 column, 10 column, 10 column
#12345678901234567890123456789
hello_____world_____foobar____
he________wo________foo_______
```

**Comment by super-tomcat** [comment]
Yes, that would be great, also the same for alignRight, thanks

**Comment by jiro4989** [comment]
@super-tomcat 

I released a new version `v1.2.1` .

added `width` optional parameter.

- [alignment API documents](https://jiro4989.github.io/alignment/alignment.html#alignLeft%2CopenArray%5Bstring%5D%2Cstring%2Cstring)

**Comment by super-tomcat** [comment]
Hello, thank you, but having tried it, it is not quite still what i asked for, hopefully this image will explain it....

![alignment](https://user-images.githubusercontent.com/30247008/129020513-d90000d7-e970-4bed-84e4-1b141b7bfb71.jpg)

**Comment by jiro4989** [comment]
Like this ?

```nim
import alignment
import strutils

let
  f1 = ["Fred Jones", "Harry Macintyre", "ニール・ヒギンズ"]
  f2 = ["45", "28", "五十六"]
  f3 = ["5.6", "5.11", "五"]

  ff1 = f1.alignLeft
  ff2 = f2.alignLeft
  ff3 = f3.alignLeft

for i, x in ff1:
  let pad = " ".repeat(10)
  echo ff1[i] & pad & ff2[i] & pad & ff3[i]
```

```bash
$ nim c a.nim
$ ./a | sed -E "s/ /_/g" | awk 'BEGIN{print "~~~~~~~~~~~~~~~~0123456789~~~~~~0123456789"}{print}'
```

result

![image](https://user-images.githubusercontent.com/13825004/129041559-801285af-cbf6-4e51-9ef8-af0b12b19fb4.png)

**Comment by super-tomcat** [comment]
Yes, that is it! Excellent 👍

**Comment by jiro4989** [comment]
Try using this development feature.

## Install development feature

```bash
$ git clone https://github.com/jiro4989/alignment
$ cd alignment
$ git checkout "feature/#2/padding"
$ nimble install -Y
```

## Use new feature

Run this code, or try your code.
I will merge this feature into master branch when this feature is that you hoped one.

```nim
import alignment

let
  f1 = ["Fred Jones", "Harry Macintyre", "ニール・ヒギンズ"]
  f2 = ["45", "28", "五十六"]
  f3 = ["5.6", "5.11", "五"]

  ff1 = f1.alignLeft(additionalPadWidth = 10)
  ff2 = f2.alignLeft(additionalPadWidth = 10)
  ff3 = f3.alignLeft(additionalPadWidth = 10)

for i, x in ff1:
  echo ff1[i] & ff2[i] & ff3[i]
```

**Comment by super-tomcat** [comment]
Yes it works fine now, thank you, my sample data would be stored in one sequence so i have to extract each row before aligning but other than that, excellent.... here is my test....

![alignment_test](https://user-images.githubusercontent.com/30247008/129100512-4e568621-ba79-468b-8c07-146560f467b5.jpg)

**Comment by jiro4989** [comment]
I released a new version `v1.3.0` .

**Comment by super-tomcat** [comment]
Great stuff, cheers 👍
